roofit/roostats/src/AsymptoticCalculator.cxx

```diff
@@ -928,24 +928,23 @@ void AsymptoticCalculator::FillBins(const RooAbsPdf & pdf, const RooArgList &obs
 
 bool AsymptoticCalculator::SetObsToExpected(RooProdPdf &prod, const RooArgSet &obs)
 {
-    RooLinkedListIter  iter(prod.pdfList().iterator());
-    bool ret = true;
-    for (RooAbsArg *a = (RooAbsArg *) iter.Next(); a != 0; a = (RooAbsArg *) iter.Next()) {
+    bool ret = true; 
+    for (auto *a : prod.pdfList()) {
         if (!a->dependsOn(obs)) continue;
         RooPoisson *pois = 0;
         RooGaussian * gaus = 0;
-        if ((pois = dynamic_cast<RooPoisson *>(a)) != 0) {
+        if ((pois = static_cast<RooPoisson *>(a)) != 0) {
```

Comment:
```suggestion
        if ((pois = dynamic_cast<RooPoisson *>(a)) != 0) {
```
Still, in the same line if checks for `!= 0` so the dynamic_cast was not superfluous here.